lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -243,7 +242,7 @@ def create_post_for_auth_or_purchase(money, payment, options)
           add_customer_data(post, options)
           add_metadata(post, options)
           post[:description] = options[:description]
-          post[:statement_description] = options[:statement_description]
+          post[:statement_descriptor] = options[:statement_description]
```

Comment:
I'm thinking that it's the responsibility of active merchant to make sure things are in sync.  So the default version specified here:

https://github.com/Shopify/active_merchant/pull/1678/files#diff-8299be227de5ba32c34c30c42a828c9cR397

should be in sync with any options that are passed through (like `statement_descriptor`, `application_fee`, `external_id`, etc).

If, in the future, Stripe adds a new API version wherein `application_fee` is renamed to `super_application_fee`, this PR would still work since we wouldn't be using that latest version, we'd still be using `2015-04-07`.  At some point someone could update Active Merchant to the latest, but then they'd need to use `super_application_fee`.

I think remote tests for Stripe are really important here.  As more options are passed, it'll be more important that we have remote tests for them.

I'd also note that overriding the `version` when you're using an `ActiveMerchant::StripeGateway` is risky since you could specify a version where the options we're passing don't match up.